libs/ui/src/hooks/useHardware.ts

```diff
@@ -0,0 +1,175 @@
+        setHasCardReaderAttached(devices.some(isCardReader));
+        setHasAccessibleControllerAttached(
+          devices.some(isAccessibleController)
+        );
+        setHasPlustekScannerAttached(devices.some(isPlustekVTM300Scanner));
+        setHasFujitsuScannerAttached(devices.some(isFujitsuScanner));
+        setAllDevices(devices);
+      });
+
+    const printerStatusSubscription = hardware.printers
+      .pipe(map((printers) => Array.from(printers)))
+      .subscribe(async (printers) => {
+        const newHasPrinterAttached = printers.some(
+          ({ connected }) => connected
+        );
+        setHasPrinterAttached(newHasPrinterAttached);
+        setAllPrinters(printers);
+      });
+    return () => {
+      hardwareStatusSubscription.unsubscribe();
+      printerStatusSubscription.unsubscribe();
+    };
+  }, [hardware]);
+
+  // Handle logging of printers
+  useEffect(() => {
+    for (const newPrinter of allPrinters) {
+      const previousCopyOfPrinter =
+        previousPrinters &&
+        previousPrinters.find((printer) => printer.name === newPrinter.name);
+      if (!previousCopyOfPrinter) {
+        void logger.log(LogEventId.PrinterAttached, 'system', {
+          message: `New printer attached: ${newPrinter.name} with connection status: ${newPrinter.connected}`,
+          printer: newPrinter.name,
+          connected: newPrinter.connected.toString(),
+        });
+      } else if (previousCopyOfPrinter.connected !== newPrinter.connected) {
+        void logger.log(LogEventId.PrinterConnectionUpdate, 'system', {
+          message: `Printer ${newPrinter.name} has been ${
+            newPrinter.connected ? 'connected' : 'disconnected'
+          }.`,
+          printer: newPrinter.name,
+          connected: newPrinter.connected.toString(),
+        });
+      }
+    }
+    for (const oldPrinter of previousPrinters || []) {
+      const newCopyOfPrinter = allPrinters.find(
+        (printer) => printer.name === oldPrinter.name
+      );
+      if (!newCopyOfPrinter) {
+        void logger.log(LogEventId.PrinterUnattached, 'system', {
+          message: `Printer unattached: ${oldPrinter.name}}`,
+          printer: oldPrinter.name,
+        });
+      }
+    }
+  }, [previousPrinters, allPrinters, logger]);
+
+  // Handle logging of devices
+  useEffect(() => {
+    for (const newDevice of allDevices) {
+      const previousCopyOfDevice =
+        previousDevices &&
+        previousDevices.find(
+          (device) =>
+            device.productId === newDevice.productId &&
+            device.vendorId === newDevice.vendorId
+        );
+      if (!previousCopyOfDevice) {
+        void logger.log(LogEventId.DeviceAttached, 'system', {
+          message: `New ${getDeviceName(newDevice)} attached. Vendor: ${
+            newDevice.vendorId
+          } , Product: ${newDevice.productId}`,
+          productId: newDevice.productId.toString(),
+          vendorId: newDevice.vendorId.toString(),
```

Comment:
These are typically displayed as hex numbers, in my experience.

```suggestion
          productId: newDevice.productId.toString(16),
          vendorId: newDevice.vendorId.toString(16),
```